### PR TITLE
Demo json nullable

### PR DESCRIPTION
**Pull request by james-baker-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/ContactController.kt [hunk 1]
```diff
@@ -128,6 +130,51 @@ class ContactController(val contactService: ContactService) {
...
+  )
+  @PreAuthorize("hasAnyRole('ROLE_CONTACTS_ADMIN')")
+  fun patchContact(
+    @PathVariable("contactId") @Parameter(
+      name = "contactId",
+      description = "The id of the contact",
+      example = "123456",
+    ) contactId: Long,
+    @Valid @RequestBody request: PatchContactRequest,
+  ): ResponseEntity<Any> {
```
Comment (james-baker-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/resource/ContactController.kt:162]: Just returns 3 different codes for the purposes of demoing it. This logic would be moved to the service layer and update the correct fields.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/PatchContactRequest.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.hmppscontactsapi.model.request
+
+import io.swagger.v3.oas.annotations.media.Schema
+import jakarta.validation.constraints.Size
+import org.openapitools.jackson.nullable.JsonNullable
+
+class PatchContactRequest {
```
Comment (james-baker-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/PatchContactRequest.kt:7]: The weak point is we can't use data class. It has to be initialised to undefined and any mandatory vars need to be lateinit which is not ideal.

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/PatchContactRequest.kt:7]: There is a broadly similar approach taken here - https://techblog.dac.digital/sealed-classes-to-rescue-the-patch-endpoints-in-kotlin-d5aa7c3e02a6 - using sealed classes, which does still retain the data class for the request, but has another downside - requiring customer JSON mappers. All solutions will be similar - allowing the 3-states on request (present and null, present and not null, not present). Swings and roundabouts which method to use..

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppscontactsapi/model/request/PatchContactRequest.kt:7]: Your example here is clear enough, and does the job.

**Pull request closed without merging**